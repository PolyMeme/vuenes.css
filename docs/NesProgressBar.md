# nes-progress-bar 

Vue Component implementation of the Nes Progress Bar 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## props 

- `value` ***Number*** (*optional*) `default: 0` 

  Define the current value of the progress bar 

- `max` ***Number*** (*optional*) `default: 1.0` 

  Define the max value of the progress bar 

- `primary` ***Boolean*** (*optional*) `default: false` 

  Define if the progress bar is using the `is-primary` css class 

- `success` ***Boolean*** (*optional*) `default: false` 

  Define if the progress bar is using the `is-success` css class 

- `warning` ***Boolean*** (*optional*) `default: false` 

  Define if the progress bar is using the `is-warning` css class 

- `error` ***Boolean*** (*optional*) `default: false` 

  Define if the progress bar is using the `is-error` css class 

- `pattern` ***Boolean*** (*optional*) `default: false` 

  Define if the progress bar is using the `is-pattern` css class 

## computed properties 

- `progressBarClasses` 

  Computed Property that calculate the css classes depending of the props used. 

   **dependencies:** `primary`, `success`, `warning`, `error`, `pattern` 


