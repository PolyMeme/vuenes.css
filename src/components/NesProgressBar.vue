<template>
  <progress
    class="nes-progress"
    :class="progressBarClasses"
    :value="value"
    :max="max"
  />
</template>

<script>
export default {
  props: {
    value: {
      type: Number,
      default: 0
    },
    max: {
      type: Number,
      default: 1.0
    },
    primary: Boolean,
    success: Boolean,
    warning: Boolean,
    error: Boolean,
    pattern: Boolean
  },
  computed: {
    progressBarClasses () {
      return {
        'is-primary': this.primary,
        'is-success': this.success,
        'is-warning': this.warning,
        'is-error': this.error,
        'is-pattern': this.pattern
      }
    }
  }
}
</script>

<style lang="scss">
@import "~typeface-press-start-2p/index.css";
@import "~nes.css/scss/base/index";
@import "~nes.css/scss/elements/progress";
</style>
