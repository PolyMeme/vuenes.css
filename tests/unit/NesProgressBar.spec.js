import { mount } from '@vue/test-utils'
import NesProgressBar from '../../src/components/NesProgressBar.vue'

describe('NesProgressBar.vue', () => {
  const createProgressBar = (propsData) => {
    return mount(NesProgressBar, {
      propsData
    })
  }

  it('should render bar properly', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100
    })

    expect(wrapper.find('progress.nes-progress').exists()).toBeTruthy()
  })

  it('should add \'is-primary\' class if prop primary is true', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100,
      primary: true
    })

    expect(wrapper.find('progress.nes-progress.is-primary').exists()).toBeTruthy()
  })

  it('should add \'is-success\' class if prop success is true', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100,
      success: true
    })

    expect(wrapper.find('progress.nes-progress.is-success').exists()).toBeTruthy()
  })

  it('should add \'is-warning\' class if prop warning is true', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100,
      warning: true
    })

    expect(wrapper.find('progress.nes-progress.is-warning').exists()).toBeTruthy()
  })

  it('should add \'is-error\' class if prop error is true', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100,
      error: true
    })

    expect(wrapper.find('progress.nes-progress.is-error').exists()).toBeTruthy()
  })

  it('should add \'is-pattern\' class if prop pattern is true', () => {
    const wrapper = createProgressBar({
      value: 50,
      max: 100,
      pattern: true
    })

    expect(wrapper.find('progress.nes-progress.is-pattern').exists()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const wrapper = createProgressBar({
      value: 80,
      max: 100,
      pattern: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
