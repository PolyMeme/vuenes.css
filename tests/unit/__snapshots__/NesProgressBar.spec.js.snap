// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesProgressBar.vue should match snapshot 1`] = `<progress max="100" class="nes-progress is-pattern"></progress>`;
